lib/services/oauth2_service.ex

```diff
@@ -50,6 +55,58 @@ defmodule EaRestaurantDataLoader.Lib.Services.Oauth2Service do
     }
   end
 
+  def login(%{
+        grant_type: "PASSWORD",
+        client_id: client_id,
+        client_secret: client_secret,
+        username: username,
+        password: password
+      }) do
+    secret_key = Application.get_env(:ea_restaurant_data_loader, :secret_key)
+
+    client = get_client_by_client_id_and_entity_status(client_id, Status.active())
+
+    validate_client_credentials(client, client_secret)
+
+    [client_scope | _] = client.scopes
+
+    user = get_user_by_username_and_entity_status(username, Status.active())
+
+    validate_user_credentials(user, password)
+
+    access_token =
+      Oauth2Util.build_token(%{grant_type: "PASSWORD"}, %{
```

Comment:
Just send the `grant_type` coming from `login(...)`

Reply:
Done.